Framework/DataObjects/inc/MantidDataObjects/MDEvent.h

```diff
@@ -45,13 +45,17 @@ template <size_t nd> class DLLExport MDEvent : public MDLeanEvent<nd> {
   /** Detector ID of the pixel that measured this event. */
   int32_t detectorId;
 
+  /** 0-based index defining the goniometer settings when the event took place*/
+  uint16_t goniometerIndex;
```

Comment:
Not doing a full review of this (just saw it by chance), but you want this to be next to `runIndex`, which is also 16 bit. As it is written now you probably end up using 3 * 32 bit since for alignment reasons there will be padding of the 16 bit fields to 32 bit. If you move the `goniometerIndex` before `detectorId` it should be packed into 2 * 32 bit.

Reply:
Thanks for pointing this out. I did  a quick calculation and the padding increases the memory footprint of `MDEvent` instances by about 9% so I will reorder the attributes as you indicated 😃